src/application/event_loop.rs

```diff
@@ -429,6 +431,115 @@ impl<T: 'static> EventLoop<T> {
               }
               Inhibit(false)
             });
+
+            let tx_clone = event_tx.clone();
+            window.connect_enter_notify_event(move |_, _| {
+              if let Err(e) = tx_clone.send(Event::WindowEvent {
+                window_id: id,
+                event: WindowEvent::CursorEntered {
+                  // FIXME: currently we use a dummy device id, find if we can get device id from gtk
```

Comment:
Is it possible to `get_device` from closure's parameter and then get either vendor id or product id?

Reply:
yeah it is possible to get the device vendor id or product id but it returns a `GString` and I have no idead what is that and winit expects a `u32`

Reply:
The field of DeviceID is not public. I think you are free to change to any type you want. Be sure to implement Copy and Clone trait. I know this is usually not recommended but compatibility with winit API is more important to us.

Reply:
Or we can hash the string into an integer.

Reply:
`get_product_id` and `get_vendor_id` seems to be behind a feature flag `#[cfg(any(feature = "v3_16", feature = "dox"))]`

and btw winit is also passing an empty `DeviceID` on some of their backends like windows so I think we should be fine.